dev/com.ibm.ws.jdbc_fat/test-applications/dsdfat_override_lib/src/dsdfat_override_lib/DSDOverrideLibTestServlet.java

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package dsdfat_override_lib;
+
+import javax.annotation.Resource;
+import javax.annotation.sql.DataSourceDefinition;
+import javax.sql.DataSource;
+
+import componenttest.app.FATServlet;
+
+@SuppressWarnings("serial")
+@DataSourceDefinition(name = "java:comp/env/jdbc/annoDSOverride",
+                      className = "org.apache.derby.jdbc.EmbeddedXADataSource40",
+                      databaseName = "memory:dsdfat_override_lib",
+                      properties = { "createDatabase=create" })
+public class DSDOverrideLibTestServlet extends FATServlet {
+
+    @Resource(lookup = "java:comp/env/jdbc/annoDSOverride")
+    DataSource annoDS;
+
+    @Resource(lookup = "java:app/jdbc/xmlDSOverride")
+    DataSource xmlDS;
+
+    public void testDataSourceDefOverrideLib() throws Exception {
+        annoDS.getConnection().close();
+        xmlDS.getConnection().close();
```

Comment:
```suggestion
        try (Connection con = annoDS.getConnection()) {
        }
        try (Connection con = xmlDS.getConnection()) {
        }
```
suggestion: we try to keep all getConnection requests in a try-with-resources block to put into practice what we tell customers to do to avoid leaked connections (even though it's not necessary here)

Reply:
I'll push up another change to the existing class `dsdfat_global_lib.DSDGlobalLibTestServlet` that I copied this one from to be consistent.